### PR TITLE
Support Gradle test patterns in the test runner (KT-30533)

### PR DESCRIPTION
**Pull request by ilmat192**

### CODE REVIEW

File: runtime/src/main/kotlin/kotlin/native/internal/test/TestStatistics.kt [hunk 1]
```diff
@@ -26,26 +26,44 @@ internal class MutableTestStatistics: TestStatistics {
     override var totalSuites: Int = 0; private set
 
     override val failed: Int
-        get() = failedTests_.size
+        get() = _failedTests.size
```
Comment (olonho) [inline runtime/src/main/kotlin/kotlin/native/internal/test/TestStatistics.kt]: Why _ before? Generally _ after gives better tooling hints.

Comment (ilmat192) [reply runtime/src/main/kotlin/kotlin/native/internal/test/TestStatistics.kt]: According to the coding conventions: https://kotlinlang.org/docs/reference/coding-conventions.html#property-names.

**Comment by snrostov** [comment]
Is there any issue about stacktraces to track it?

https://github.com/JetBrains/kotlin-native/blob/5a9aac55da52db0395c8f84c3c5f251b9df050be/runtime/src/main/kotlin/kotlin/native/internal/test/TeamCityLogger.kt#L54

File: runtime/src/main/kotlin/kotlin/native/internal/test/TestStatistics.kt (hunk 1 above)
Comment (olonho) [reply runtime/src/main/kotlin/kotlin/native/internal/test/TestStatistics.kt]: Hmm, maybe coding convention better be rethought here. Anyway, not in this PR.

**Pull request merged**